lib/refile/custom_logger.rb

```diff
@@ -31,5 +29,14 @@ def log(env, status, began_at)
         (now - began_at) * 1000
       )
     end
+
+    def logger
+      @logger ||= @logger_proc.call
```

Comment:
Worst case scenario would be that multiple threads would run `proc { Refile.logger }` multiple times, but `Refile.logger` is still the same object. I don't see any way how this would harm anything.